inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/configuration/http/HttpConfigurationFactory.java

```diff
@@ -40,17 +48,17 @@ private static SimpleRequestBuilder buildRequestWithHeaders(
       SimpleRequestBuilder requestBuilder, Agent agent) {
     Map<String, String> headers =
         Map.of(
-            "x-gepard-service-Name", agent.getServiceName(),
-            "x-gepard-vm-id", agent.getVmId(),
-            "x-gepard-gepard-version", agent.getGepardVersion(),
-            "x-gepard-otel-version", agent.getOtelVersion(),
-            "x-gepard-java-version", agent.getJavaVersion(),
-            "x-gepard-start-time", agent.getStartTime().toString());
+            X_GEPARD_SERVICE_NAME, agent.getServiceName(),
+            X_GEPARD_VM_ID, agent.getVmId(),
+            X_GEPARD_GEPARD_VERSION, agent.getGepardVersion(),
+            X_GEPARD_OTEL_VERSION, agent.getOtelVersion(),
+            X_GEPARD_JAVA_VERSION, agent.getJavaVersion(),
+            X_GEPARD_START_TIME, agent.getStartTime().toString());
 
     headers.forEach(requestBuilder::addHeader);
     agent
         .getAttributes()
-        .forEach((key, value) -> requestBuilder.addHeader("X-Gepard-Attribute-" + key, value));
+        .forEach((key, value) -> requestBuilder.addHeader(X_GEPARD_ATTRIBUTE + key, value));
```

Comment:
Hi @EddeCCC, @binarycoded,
Quick note: I changed the headers to lower-case, since the apache http client automatically lowercases them, when sending.
According to the http-specs, header names are case-insensitive (https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive). That might be the reason. 
Long Story Short: On the the other side (config-server) you still need to check for lower-cased headers. 
Was pretty painful to find that out while debugging.